auraed/src/cri/runtime_service.rs

```diff
@@ -115,72 +125,111 @@ impl runtime_service_server::RuntimeService for RuntimeService {
-            .with_root_path(Path::new(AURAE_PODS_PATH).join(sandbox_id.clone()))
-            .expect("Setting pods directory")
-            .as_init(Path::new(AURAE_BUNDLE_PATH).join(AURAE_SELF_IDENTIFIER))
-            .with_systemd(false)
-            .build()
-            .expect("failed building pod sandbox: ensure valid OCI spec and proper container starting point");
-
-        sandbox.start().expect("starting pod sandbox");
+        let mut sandbox: Option<Container>;
+        {
+            // Initialize a new container builder with the AURAE_SELF_IDENTIFIER name as the "init" container running a recursive Auraed
+            let syscall = create_syscall();
+            let sandbox_builder = ContainerBuilder::new(
+                AURAE_SELF_IDENTIFIER.to_string(),
+                syscall.as_ref(),
+            );
+
+            // Spawn auraed here
+            // TODO Check if sandbox already exists and if it does, clean up to this point?
+            let _spawned = spawn_auraed_oci_to(
+                Path::new(AURAE_BUNDLE_PATH)
+                    .join(AURAE_SELF_IDENTIFIER)
+                    .to_str()
+                    .expect("recursive path"),
+                oci_builder.build().expect("building pod oci spec"),
+            );
+
+            sandbox = Some(sandbox_builder
+                .with_root_path(Path::new(AURAE_PODS_PATH).join(sandbox_id.clone()))
+                .expect("Setting pods directory")
+                .as_init(Path::new(AURAE_BUNDLE_PATH).join(AURAE_SELF_IDENTIFIER))
+                .with_systemd(false)
+                .build()
+                .expect("failed building pod sandbox: ensure valid OCI spec and proper container starting point"));
+
+            sandbox
+                .as_mut()
+                .expect("a sandbox")
+                .start()
+                .expect("starting pod sandbox");
+        }
 
-        // TODO: Cache sandbox
+        // NOTE: we're creating a libcontainer::container and calling it `sandbox`,
+        // so that's what goes in the cache.  but i think the rest of this service
+        // differentiates between the outer "pod" and the inner "container"s so we
+        // might want to do the same.
+        let mut sandboxes = self.sandboxes.lock().await;
+        sandboxes.add(sandbox_id.clone(), sandbox.expect("a sandbox"))?;
 
         Ok(Response::new(RunPodSandboxResponse { pod_sandbox_id: sandbox_id }))
     }
 
     async fn stop_pod_sandbox(
         &self,
-        _request: Request<StopPodSandboxRequest>,
+        request: Request<StopPodSandboxRequest>,
     ) -> Result<Response<StopPodSandboxResponse>, Status> {
-        // TODO: Pull sandbox from cache
-        // TODO: sandbox.kill()
-        todo!()
+        let sandbox_id = request.into_inner().pod_sandbox_id;
+
+        let mut sandboxes = self.sandboxes.lock().await;
+        let sandbox = sandboxes.get_mut(&sandbox_id)?;
+        sandbox.kill(Signal::SIGKILL, false).map_err(|e| {
```

Comment:
Should we not be removing it from the cache after the kill?

Reply:
i thought about that but there's an explicit `remove_pod_sandbox` call just below, so i thought that might be the equivalent of "stop" and "free".